### DISCUSSION TITLE
How to force explicit type declerations.

### DISCUSSION THREAD

**Discussion by urosjarc** [detekt/detekt#5783] [Q&A] [answered]
For readability purposes, I prefer to write...

    val value: Int = 0

...rather than...

    val value = 0

...but sometimes, I forget to add the type declaration, which leads to inconsistent code style.

Is there an option to enforce explicit type declarations?

**Comment by schalkms** [comment, marked as answer]
You could write a [custom rule ](https://detekt.dev/docs/introduction/extensions#custom-rulesets)that flags this scenarios.

To view the AST (PSI) of your source code, you can use the [PSI Viewer plugin](https://plugins.jetbrains.com/plugin/227-psiviewer) for IntelliJ. This is helpful when implementing and debugging rules.

**Comment by BraisGabin** [comment]
If you only care about that on the public functions/properties you could also use this rule: https://detekt.dev/docs/rules/libraries#librarycodemustspecifyreturntype
